test/modern_bpf/test_suites/syscall_exit_suite/connect_x.cpp

```diff
@@ -0,0 +1,241 @@
+#include "../../event_class/event_class.h"
+
+#if defined(__NR_connect) && defined(__NR_socket) && defined(__NR_bind) && defined(__NR_close)
+
+#include <netdb.h>
+#include <sys/un.h>
+
+TEST(SyscallExit, connectX_INET)
```

Comment:
Hi @Andreagit97,
running the test suite, those tests are failing:
````
[  FAILED  ] SyscallExit.connectX_INET
[  FAILED  ] SyscallExit.connectX_INET6
[  FAILED  ] SyscallExit.connectX_UNIX
````

Here are few failures... uploading the others as logfile:
````
/root/git/falcosecurity/libs/test/modern_bpf/event_class/event_class.cpp:391: Failure
Expected equality of these values:
  family
    Which is: '\0'
  desired_family
    Which is: '\x2' (2)
>>>>> value of the param is not correct. Param id = 2

/root/git/falcosecurity/libs/test/modern_bpf/event_class/event_class.cpp:408: Failure
Expected equality of these values:
  ipv4_string
    Which is: "0.0.0.0"
  desired_ipv4
    Which is: "127.0.21.34"
>>>>> value of the param is not correct. Param id = 2 (source ipv4)

[...]

[  FAILED  ] SyscallExit.connectX_INET (0 ms)
[ RUN      ] SyscallExit.connectX_INET6
/root/git/falcosecurity/libs/test/modern_bpf/event_class/event_class.cpp:391: Failure
Expected equality of these values:
  family
    Which is: '\0'
  desired_family
    Which is: '\n' (10, 0xA)
With diff:
@@ -1,1 +1,2 @@
-'\0'
+'
+' (10, 0xA)

>>>>> value of the param is not correct. Param id = 2
````
[connectX-failures.log](https://github.com/falcosecurity/libs/files/9309175/connectX-failures.log)

Reply:
Hey @hbrueckner thank you very much for this, i will take a look ASAP it seems the kernel doesn't find any data while we expect the tuple to be filled :thinking: The strange thing is that on `x86` and `aarch64` it works :thinking: Does it fails every time or just in some runs?

Reply:
Hi @Andreagit97 , thanks for looking into this. The tests fails on every test run (even after reboot).

Reply:
Looks indeed that getting the `socket_family` seems to fail in:
``
Here is what I tried:
````diff
--- a/driver/modern_bpf/helpers/store/auxmap_store_params.h
+++ b/driver/modern_bpf/helpers/store/auxmap_store_params.h
@@ -612,6 +612,9 @@ static __always_inline void auxmap__store_socktuple_param(struct auxiliary_map *
        struct sock *sk = BPF_CORE_READ(socket, sk);
        BPF_CORE_READ_INTO(&socket_family, socket, ops, family);
 
+
+       bpf_printk("socket_family: %u\n", socket_family);
+
        switch(socket_family)
        {
        case AF_INET:
````
resulting in:
````
[root@m8345024 build]# cat /sys/kernel/debug/tracing/trace
# tracer: nop
#
# entries-in-buffer/entries-written: 3/3   #P:2
#
#                                _-----=> irqs-off/BH-disabled
#                               / _----=> need-resched
#                              | / _---=> hardirq/softirq
#                              || / _--=> preempt-depth
#                              ||| / _-=> migrate-disable
#                              |||| /     delay
#           TASK-PID     CPU#  |||||  TIMESTAMP  FUNCTION
#              | |         |   |||||     |         |
        bpf_test-3015    [000] d....  1201.646247: bpf_trace_printk: socket_family: 0

        bpf_test-3015    [000] d....  1201.646630: bpf_trace_printk: socket_family: 0

        bpf_test-3015    [000] d....  1201.646992: bpf_trace_printk: socket_family: 0
````